bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -848,6 +848,9 @@ private void removeMissingModuleTypes(Collection<String> moduleTypes) {
             if (rules != null) {
                 for (String rUID : rules) {
                     RuleStatus ruleStatus = getRuleStatus(rUID);
+                    if (ruleStatus == null) {
```

Comment:
If we already knows that ruleStatus is null, we should not call `switch (ruleStatus)` after that. This is a grant for a NPE.